File: script-gen-ui/src/app/model/sweep_data/statusMsg.ts
import { IStatusMsg, StatusType } from '../interface';

export class StatusMsg {
  status_type: StatusType;
  message: string;
  time_stamp: string;

  constructor(data: IStatusMsg) {
    this.status_type = data.status_type;
    this.message = data.message;
    this.time_stamp = data.time_stamp;
  }
}

File: script-gen-manager/src/model/sweep_data/status_msg.rs
use chrono::Local;
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
pub struct StatusMsg {
    pub status_type: StatusType,
    pub message: String,
    pub time_stamp: String,
}

#[derive(Debug, Clone, Serialize)]
pub enum StatusType {
    Info,
    Warning,
    Error,
}

impl StatusMsg {
    #[must_use]
    pub fn new(status_type: StatusType, message: String) -> Self {
        let time_stamp = Local::now().format("%Y-%m-%d %H:%M:%S").to_string();
        Self {
            status_type,
            message,
            time_stamp,
        }
    }
}
